library/src/main/java/com/timehop/stickyheadersrecyclerview/HeaderPositionCalculator.java

```diff
@@ -19,6 +19,9 @@
   private final OrientationProvider mOrientationProvider;
   private final HeaderProvider mHeaderProvider;
   private final DimensionCalculator mDimensionCalculator;
+  
+  private final Rect bufferRect1 = new Rect();
+  private final Rect bufferRect2 = new Rect();
```

Comment:
My intent was to leave these variable context-less, explicitly showing that they are just buffers which are useful only in context of particular method. The names that you proposed will definitely work, but if, for example, we decide to calculate footer rect we'll have to write something `initFooterRect(mHeaderRect, ...); float footerX = mHeaderRect.x;` which will be more confusing than `initFooterRect(mTempRect, ...); float footerX = mTempRect.x;`. There is another aspect: a future reader might think that `mHeaderRect` is some useful permanent field, but it will just contain some leftovers from previous calculations. `mTmpRect` suggests the latter. 
Just my thoughts...